allennlp/common/cached_transformers.py

```diff
@@ -75,11 +75,17 @@ def strip_prefix(s):
                     )
                 override_weights = {strip_prefix(k): override_weights[k] for k in valid_keys}
 
-            transformer = AutoModel.from_pretrained(
-                model_name,
-                state_dict=override_weights,
-                **kwargs,
+            transformer = AutoModel.from_config(
+                AutoConfig.from_pretrained(
+                    model_name,
+                    **kwargs,
+                )
             )
+            # Only load the model itself if we are using distributed training
```

Comment:
Can you explain this more? I'm not sure what would different in distributed training

Reply:
https://github.com/allenai/allennlp/pull/5085/#discussion_r607213865